wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/jpa/JpaRecipeDao.java

```diff
@@ -101,33 +104,60 @@ public RecipeImpl getById(String id) throws NotFoundException, ServerException {
     }
   }
 
+  /**
+   * Translated query should look like:
+   *
+   * <p>SELECT recipe.ID, recipe.CREATOR, recipe.DESCRIPTION, recipe.NAME, recipe.SCRIPT,
```

Comment:
IMHO it is really difficult to read this query. Please reformat it

Reply:
I can not due to fmt plugin :(

Reply:
Formatting in this way works fine for me
```
/**
   * Translated query should look like:
   *
   * <pre>
   * SELECT recipe.ID, recipe.CREATOR, recipe.DESCRIPTION, 
   *     recipe.NAME, recipe.SCRIPT, recipe.TYPE
   * FROM {oj RECIPEPERMISSIONS permission
   * LEFT OUTER JOIN RECIPE recipe ON (recipe.ID = permission.RECIPEID)
   * LEFT OUTER JOIN Recipe_TAGS tag ON (tag.Recipe_ID = recipe.ID)}, 
   *     RECIPEPERMISSIONS_ACTIONS permissionActions
   * WHERE ((tag.tag IN (?)) 
   *     AND ((? IS NULL) OR (recipe.TYPE IS NULL) OR (recipe.TYPE = ?)) 
   *     AND ((permission.USERID IS NULL) OR (permission.USERID = ?)) 
   *     AND (permissionActions.actions = ?) 
   *     AND (permissionActions.RECIPEPERMISSIONS_ID = permission.ID))
   * GROUP BY recipe.ID 
   * HAVING (COUNT(tag.tag) = ?)
   * </pre>
   */
```